src/styles/main.css

```diff
@@ -0,0 +1,199 @@
+}
+
+.calendar__day--start-day-tue :first-child {
+  margin-left: 101px;
+}
+
+.calendar__day--start-day-wed :first-child {
+  margin-left: 202px;
+}
+
+.calendar__day--start-day-thu :first-child {
+  margin-left: 303px;
+}
+
+.calendar__day--start-day-fri :first-child {
+  margin-left: 404px;
+}
+
+.calendar__day--start-day-sat :first-child {
+  margin-left: 505px;
+}
+
+.calendar__day--start-day-sun :first-child {
+  margin-left: 606px;
+}
+
+.calendar__day::before {
+  justify-content: center;
+  content: counter(day);
+  counter-increment: day;
+  display: flex;
+}
+
+.calendar__day:hover {
+  cursor: pointer;
+  background-color: #ffbfcb;
+  transform: translateY(-20px);
+}
+
+.calendar--start-day-sunday .calendar__day:nth-child(-n + 6) {
+  margin-left: 0;
+}
+
+.calendar--month-length:nth-child(1)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(2)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(3)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(4)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(5)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(6)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(7)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(8)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(9)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(10)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(11)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(12)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(13)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(14)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(15)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(16)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(17)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(18)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(19)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(20)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(21)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(22)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(23)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(24)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(25)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(26)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(27)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(28)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(29)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(30)::before {
+  display: none;
+}
+
+.calendar--month-length:nth-child(31)::before {
+  display: none;
```

Comment:
The use of `nth-child` selectors to hide elements seems excessive and may not be efficient. Consider using a more concise method, such as a class that can be toggled, to manage visibility.